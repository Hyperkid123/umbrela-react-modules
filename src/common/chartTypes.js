export const PIE_CHART = 'PIE_CHART';
export const BAR_CHART = 'BAR_CHART';
export const POLAR_CHART = 'POLAR_CHART';
export const LINE_CHART = 'LINE_CHART';
export const DOUGHNUT_CHART = 'DOUGHNUT_CHART';

export const OPEN_QUESTION = 'OpenQuestion';
export const OPEN_WITH_IMAGE_QUESTIONS = 'OpenWithImageQuestion';

export const CLOSE_QUESTION = 'CloseQuestion';
export const CLOSE_MULTI_QUESTION = 'CloseMultiQuestion';
export const CLOSE_WITH_OPEN_QUESTION = 'CloseWithOpenQuestion';
export const CLOSE_MULTI_WITH_OPEN_QUESTION = 'CloseMultiWithOpenQuestion';

export const IMAGE_OPTIONS_QUESTION = 'ImageOptionsQuestion';
export const IMAGE_OPTIONS_MULTI_QUESTION = 'ImageOptionsMultiQuestion';
export const IMAGE_OPTIONS_WITH_OPEN_QUESTION = 'ImageOptionsWithOpenQuestion';
export const IMAGE_OPTIONS_MULTI_WITH_OPEN_QUESTION = 'ImageOptionsMultiWithOpenQuestion';

export const MEDIA_QUESTION = 'MediaQuestion';
export const MEDIA_MULTI_QUESTION = 'MediaMultiQuestion';
export const MEDIA_WITH_OPEN_QUESTION = 'MediaWithOpenQuestion';
export const MEDIA_MULTI_WITH_OPEN_QUESTION = 'MediaMultiWithOpenQuestion';

export const MATRIX_SINGLE_QUESTION = 'MatrixSingleQuestion';
export const MATRIX_MULTI_QUESTION = 'MatrixMultiQuestion';
export const MATRIX_SINGLE_IMAGE_QUESTION = 'MatrixSingleImageQuestion';
export const MATRIX_MULTI_IMAGE_QUESTION = 'MatrixMultiImageQuestion';

export const DIVIDE_QUESTION = 'DivideQuestion';
export const ORDER_QUESTION = 'OrderQuestion';

export function isOptionImage(questionType){
  return questionType === IMAGE_OPTIONS_QUESTION
  || IMAGE_OPTIONS_MULTI_QUESTION === questionType
  || questionType === IMAGE_OPTIONS_WITH_OPEN_QUESTION
  || questionType === IMAGE_OPTIONS_MULTI_WITH_OPEN_QUESTION
  || questionType === MATRIX_MULTI_IMAGE_QUESTION
  || questionType === MATRIX_SINGLE_IMAGE_QUESTION;
}

export function hasImagePreview(questionType){
  return questionType === MEDIA_QUESTION ||
  questionType === MEDIA_MULTI_QUESTION ||
  questionType === MEDIA_WITH_OPEN_QUESTION ||
  questionType === MEDIA_MULTI_WITH_OPEN_QUESTION;
}
