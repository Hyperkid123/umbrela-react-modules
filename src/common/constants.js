import {
  deepOrange500,
  grey700,
  grey100,
} from 'material-ui/styles/colors';

export const LABEL_LENGTH = 255;
export const LABEL_ERROR_STYLE = {
  textAlign: 'end',
}
export const TEXT_FIELD_UNDERLINE = {
  borderColor: grey100,
}

export const TEXT_FIELD_FOCUS = {
  borderColor: deepOrange500,
}
