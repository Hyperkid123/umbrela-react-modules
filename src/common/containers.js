import styled from 'styled-components';

export const Flex = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: ${props => props.column ? 'column' : 'row'};
  align-items:  ${props => props.verticalCenter ? 'center' : 'null'};
  justify-content: ${props => props.horizintalCenter ? 'center' : 'null'};
`
