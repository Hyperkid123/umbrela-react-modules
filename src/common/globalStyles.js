import {injectGlobal} from 'styled-components';
import deepOrange from 'material-ui/colors/deepOrange';
import grey from 'material-ui/colors/grey';

injectGlobal`
  body{
    margin: 0;
    padding: 0;
    background-color: ${grey[200]};
    font-family: 'Roboto', sans-serif;
  }
  .active-link > button, .active-link svg {
    color: ${deepOrange[500]} !important;
    fill: ${deepOrange[500]} !important
  }
  a {
    text-decoration: none;
  }
`;
