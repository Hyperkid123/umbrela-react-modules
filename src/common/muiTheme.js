import deepOrange from 'material-ui/colors/deepOrange';

import { createMuiTheme } from 'material-ui/styles';
export default createMuiTheme({
    fontFamily: 'Roboto, sans-serif',
    palette: {
      primary: {
        main: deepOrange[500],
        dark: deepOrange[500],
      },
    },
    raisedButton: {
      fontWeight: 300
    },
    appBar: {
      titleFontWeight: 300
    },
});
