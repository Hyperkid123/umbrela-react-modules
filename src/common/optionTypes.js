export const OptionTypes = {
    NormalOption: 'NormalOption',
    OpenOption: 'OpenOption',
    ColumnOption: 'ColumnOption',
    RowOption: 'RowOption',
};
