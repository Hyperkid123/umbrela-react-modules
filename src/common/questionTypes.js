export const BaseTypes = [
    'OpenQuestion',
    'CloseQuestion',
    'MatrixSingleQuestion',
    'OrderQuestion',
    'DivideQuestion',
    'MediaQuestion',
];

export const QuestionTypes = {
    OpenQuestion: 'Otevřená',
    CloseQuestion: 'Uzavřená',
    CloseWithOpenQuestion: 'Uzavřená s vlasní odpovědí',
    CloseMultiQuestion: 'Uzavřená s více možnostmi',
    CloseMultiWithOpenQuestion: 'Zavřená s více možnostmi a vlastní odpovědí',
    MatrixSingleQuestion: 'Maticová otázka',
    MatrixMultiQuestion: 'Maticová otázka s volbou více možností',
    MatrixTextQuestion: 'Maticová s texte',
    OrderQuestion: 'Seřazovací otázka',
    DivideQuestion: 'Rozdělovací otázka',
    OpenLongQuestion: 'Open long',
    OpenWithImageQuestion: 'Otevřená s obrázkem v zadání',
    MediaQuestion: 'Uzavřená s obrázkem v zadání',
    MediaMultiQuestion: 'Uzavřená s více možnostmi s obrázkem v zadání',
    MediaWithOpenQuestion: 'Uzavřená s obrázkem a vlastní odpovědí',
    MediaMultiWithOpenQuestion: 'Uzavřená s více možnostmi s obrázkem a vlastní odpovědí',
    ImageOptionsQuestion: 'Obrázky jako možnosti',
    ImageOptionsMultiQuestion: 'Obrázky jako možnosti multiple choice',
    ImageOptionsWithOpenQuestion: 'Obrázky jako možnosti s vlastní odpovědí',
    ImageOptionsMultiWithOpenQuestion: 'Obrázky jako možnosti s vlastní odpovědí multiple',
    MatrixSingleImageQuestion: 'Maticová s obrázky',
    MatrixMultiImageQuestion: 'Maticová s obrázky multiple',
    OpenWithImageQuestionLong: 'Open long s obrazkem v zadani',
};

export const CanHaveOpenQuestion = (questionType) => {
    return [
        'CloseQuestion',
        'CloseWithOpenQuestion',
        'CloseMultiQuestion',
        'CloseMultiWithOpenQuestion',
        'MediaQuestion',
        'MediaMultiQuestion',
        'MediaWithOpenQuestion',
        'MediaMultiWithOpenQuestion',
        'ImageOptionsQuestion',
        'ImageOptionsMultiQuestion',
        'ImageOptionsWithOpenQuestion',
        'ImageOptionsMultiWithOpenQuestion',
    ].includes(questionType);
};

export const HasOpenQuestion = (questionType) => {
    return [
        'CloseWithOpenQuestion',
        'CloseMultiWithOpenQuestion',
        'MediaWithOpenQuestion',
        'MediaMultiWithOpenQuestion',
        'ImageOptionsWithOpenQuestion',
        'ImageOptionsMultiWithOpenQuestion',
    ].includes(questionType);
};

export const CanHaveMultipleAnswers = (questionType) => {
    return [
        'CloseQuestion',
        'CloseWithOpenQuestion',
        'CloseMultiQuestion',
        'CloseMultiWithOpenQuestion',
        'MediaQuestion',
        'MediaMultiQuestion',
        'MediaWithOpenQuestion',
        'MediaMultiWithOpenQuestion',
        'ImageOptionsQuestion',
        'ImageOptionsMultiQuestion',
        'ImageOptionsWithOpenQuestion',
        'ImageOptionsMultiWithOpenQuestion',
        'MatrixMultiImageQuestion',
        'MatrixMultiQuestion',
        'MatrixSingleImageQuestion',
        'MatrixSingleQuestion',
    ].includes(questionType);
};

/**
 * @return {boolean}
 */
export const HasMultipleAnswers = (questionType) => {
    return [
        'CloseMultiQuestion',
        'CloseMultiWithOpenQuestion',
        'MediaMultiQuestion',
        'MediaMultiWithOpenQuestion',
        'ImageOptionsMultiQuestion',
        'ImageOptionsMultiWithOpenQuestion',
        'MatrixMultiImageQuestion',
        'MatrixMultiQuestion',
    ].includes(questionType);
};

/**
 * @return {boolean}
 */
export const IsMatrixQuestion = (questionType) => {
    return [
        'MatrixMultiImageQuestion',
        'MatrixSingleImageQuestion',
        'MatrixMultiQuestion',
        'MatrixSingleQuestion',
        'MatrixTextQuestion',
    ].includes(questionType);
};

/**
 * @return {boolean}
 */
export const HasImagePreview = (questionType) => {
    return [
        'OpenWithImageQuestion',
        'MediaQuestion',
        'MediaMultiQuestion',
        'MediaWithOpenQuestion',
        'MediaMultiWithOpenQuestion',
    ].includes(questionType);
};

export const CanHaveImagePreview = (questionType) => {
    return [
        'OpenWithImageQuestion',
        'OpenQuestion',
        'OpenLongQuestion',
        'OpenWithImageQuestionLong',
    ].includes(questionType);
};

export const HasOptionsAsImage = (questionType) => {
    return [
        'ImageOptionsQuestion',
        'ImageOptionsMultiQuestion',
        'ImageOptionsWithOpenQuestion',
        'ImageOptionsMultiWithOpenQuestion',
        'MatrixSingleImageQuestion',
        'MatrixMultiImageQuestion',

    ].includes(questionType);

};

export const HasScalePoints = (questionType) => {
    return questionType === 'DivideQuestion';
};

export const IsOrderQuestion = (questionType) => {
    return questionType === 'OrderQuestion';
};

export const HasNotOptions = (questionType) => {
    return ['OpenQuestion','OpenWithImageQuestion', 'OpenLongQuestion', 'OpenWithImageQuestionLong'].includes(questionType);
};

export const CanBeOpenLong = (questionType) => {
    return ['OpenQuestion', 'OpenLongQuestion', 'OpenWithImageQuestion', 'OpenWithImageQuestionLong'].includes(questionType);
};

export const IsLongQuestion = (questionType) => {
    return ['OpenLongQuestion', 'OpenWithImageQuestionLong'].includes(questionType);
};

export const SwitchQuestionOpenOption = {
    CloseQuestion: 'CloseWithOpenQuestion',
    CloseWithOpenQuestion: 'CloseQuestion',
    CloseMultiQuestion: 'CloseMultiWithOpenQuestion',
    CloseMultiWithOpenQuestion: 'CloseMultiQuestion',

    MediaQuestion: 'MediaWithOpenQuestion',
    MediaMultiQuestion: 'MediaMultiWithOpenQuestion',
    MediaWithOpenQuestion: 'MediaQuestion',
    MediaMultiWithOpenQuestion: 'MediaMultiQuestion',

    ImageOptionsQuestion: 'ImageOptionsWithOpenQuestion',
    ImageOptionsMultiQuestion: 'ImageOptionsMultiWithOpenQuestion',
    ImageOptionsWithOpenQuestion: 'ImageOptionsQuestion',
    ImageOptionsMultiWithOpenQuestion: 'ImageOptionsMultiQuestion',
};

export const SwitchQuestionMultipleAnswers = {
    CloseQuestion: 'CloseMultiQuestion',
    CloseWithOpenQuestion: 'CloseMultiWithOpenQuestion',
    CloseMultiQuestion: 'CloseQuestion',
    CloseMultiWithOpenQuestion: 'CloseWithOpenQuestion',

    MediaQuestion: 'MediaMultiQuestion',
    MediaMultiQuestion: 'MediaQuestion',
    MediaWithOpenQuestion: 'MediaMultiWithOpenQuestion',
    MediaMultiWithOpenQuestion: 'MediaWithOpenQuestion',

    ImageOptionsQuestion: 'ImageOptionsMultiQuestion',
    ImageOptionsMultiQuestion: 'ImageOptionsQuestion',
    ImageOptionsWithOpenQuestion: 'ImageOptionsMultiWithOpenQuestion',
    ImageOptionsMultiWithOpenQuestion: 'ImageOptionsWithOpenQuestion',

    MatrixSingleImageQuestion: 'MatrixMultiImageQuestion',
    MatrixMultiImageQuestion: 'MatrixSingleImageQuestion',
    MatrixSingleQuestion: 'MatrixMultiQuestion',
    MatrixMultiQuestion: 'MatrixSingleQuestion',
};

export const SwitchToImagePreview =  {
    OpenQuestion: 'OpenWithImageQuestion',
    OpenWithImageQuestion: 'OpenQuestion',
    OpenWithImageQuestionLong: 'OpenLongQuestion',
    OpenLongQuestion: 'OpenWithImageQuestionLong',
};

export const SwitchOptionToImage = {
    CloseQuestion: 'ImageOptionsQuestion',
    CloseWithOpenQuestion: 'ImageOptionsWithOpenQuestion',
    CloseMultiQuestion: 'ImageOptionsMultiQuestion',
    CloseMultiWithOpenQuestion: 'ImageOptionsMultiWithOpenQuestion',
    MatrixSingleQuestion: 'MatrixSingleImageQuestion',
    MatrixMultiQuestion: 'MatrixMultiImageQuestion',
    ImageOptionsQuestion: 'CloseQuestion',
    ImageOptionsMultiQuestion: 'CloseMultiQuestion',
    ImageOptionsWithOpenQuestion: 'CloseWithOpenQuestion',
    ImageOptionsMultiWithOpenQuestion: 'CloseMultiWithOpenQuestion',
    MatrixSingleImageQuestion: 'MatrixSingleQuestion',
    MatrixMultiImageQuestion: 'MatrixMultiQuestion',
};
