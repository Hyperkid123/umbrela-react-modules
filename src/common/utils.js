import {HasOpenQuestion} from './questionTypes';
import lodash from 'lodash';
import {HasMultipleAnswers} from './questionTypes';


export const getNewOptionOrder = (options, questionType) => {
  if(HasOpenQuestion(questionType)) {
    return options.length - 1;
  } else {
    return options.length;
  }
}

export const findOpenOption = (options) => {
  return lodash.find(options, (option) => {
    return option.optionType === 'OpenOption';
  });
}

export const getOptionsInputType = (questionType) => {
  return HasMultipleAnswers(questionType) ? 'checkBox' : 'redioButton'
}
