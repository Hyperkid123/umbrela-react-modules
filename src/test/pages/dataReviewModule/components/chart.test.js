import React from 'react'
import Chart from '../../../../pages/dataReviewModule/components/chart';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import * as chartTypes from '../../../../common/chartTypes'

describe('Chart component', () => {
  const initialProps = {};
  beforeEach(() => {
    initialProps.data = {};
    initialProps.chartType = 'foo';
    initialProps.showLegend = true;
    initialProps.questionType = 'foo';
    initialProps.question = {}
    initialProps.data = {barData: {}}
  })

  it('should render empty', () => {
    const tree = shallow(
      <Chart {...initialProps}/>
    );
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render open question', () => {
    const openTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.OPEN_QUESTION}
      />
    );
    expect(toJson(openTree)).toMatchSnapshot();

    const openImageTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.OPEN_WITH_IMAGE_QUESTIONS}
      />
    );
    expect(toJson(openImageTree)).toMatchSnapshot();
  })

  it('should render close question', () => {
    const closeTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.CLOSE_QUESTION}
      />
    );
    expect(toJson(closeTree)).toMatchSnapshot();

    const closeWithMultiTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.CLOSE_MULTI_QUESTION}
      />
    );
    expect(toJson(closeWithMultiTree)).toMatchSnapshot();

    const closeWithOpenTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.CLOSE_WITH_OPEN_QUESTION}
      />
    );
    expect(toJson(closeWithOpenTree)).toMatchSnapshot();

    const closeMultiWithOpenTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.CLOSE_MULTI_WITH_OPEN_QUESTION}
      />
    );
    expect(toJson(closeMultiWithOpenTree)).toMatchSnapshot();
    //image
    const imageTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.IMAGE_OPTIONS_QUESTION}
      />
    );
    expect(toJson(imageTree)).toMatchSnapshot();

    const imageMultiTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.IMAGE_OPTIONS_MULTI_QUESTION}
      />
    );
    expect(toJson(imageMultiTree)).toMatchSnapshot();

    const imageWithOpenTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.IMAGE_OPTIONS_WITH_OPEN_QUESTION}
      />
    );
    expect(toJson(imageWithOpenTree)).toMatchSnapshot();

    const imageMultiWithOpenTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.IMAGE_OPTIONS_MULTI_WITH_OPEN_QUESTION}
      />
    );
    expect(toJson(imageMultiWithOpenTree)).toMatchSnapshot();
    ///media
    const mediaTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.MEDIA_QUESTION}
      />
    );
    expect(toJson(mediaTree)).toMatchSnapshot();

    const mediaMultiTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.MEDIA_MULTI_QUESTION}
      />
    );
    expect(toJson(mediaMultiTree)).toMatchSnapshot();

    const mediaWithopenTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.MEDIA_WITH_OPEN_QUESTION}
      />
    );
    expect(toJson(mediaWithopenTree)).toMatchSnapshot();

    const mediaMultiWithOpenTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.MEDIA_MULTI_WITH_OPEN_QUESTION}
      />
    );
    expect(toJson(mediaMultiWithOpenTree)).toMatchSnapshot();
  })

  it('should render matrix question', () => {
    const matrixSingleTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.MATRIX_SINGLE_QUESTION}
      />
    );
    expect(toJson(matrixSingleTree)).toMatchSnapshot();

    const matrixMultiTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.MATRIX_MULTI_QUESTION}
      />
    );
    expect(toJson(matrixMultiTree)).toMatchSnapshot();

    const matrixSingleImageTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.MATRIX_SINGLE_IMAGE_QUESTION}
      />
    );
    expect(toJson(matrixSingleImageTree)).toMatchSnapshot();

    const matrixMultiImageTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.MATRIX_MULTI_IMAGE_QUESTION}
      />
    );
    expect(toJson(matrixMultiImageTree)).toMatchSnapshot();
  })

  it('should render order question', () => {
    const tree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.DIVIDE_QUESTION}
      />
    );
    expect(toJson(tree)).toMatchSnapshot();

    const orderTree = shallow(
      <Chart
        {...initialProps}
        questionType={chartTypes.ORDER_QUESTION}
      />
    );
    expect(toJson(orderTree)).toMatchSnapshot();
  })
});
