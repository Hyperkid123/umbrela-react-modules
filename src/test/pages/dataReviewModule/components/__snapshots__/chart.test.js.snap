// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart component should render close question 1`] = `
<WithStyles(GridWrapper)
  container={true}
>
  <WithStyles(GridWrapper)
    item={true}
    lg={6}
    md={12}
    sm={12}
    xl={6}
    xs={12}
  >
    <CloseQuestions
      chartType="foo"
      data={Object {}}
      hasImagePreview={false}
      showLegend={true}
    />
  </WithStyles(GridWrapper)>
  <WithStyles(GridWrapper)
    item={true}
    lg={6}
    md={12}
    sm={12}
    xl={6}
    xs={12}
  >
    <Connect(LegendTable)
      data={
        Object {
          "barData": Object {},
        }
      }
      isOptionImage={false}
    />
  </WithStyles(GridWrapper)>
</WithStyles(GridWrapper)>
`;

exports[`Chart component should render empty 1`] = `""`;

exports[`Chart component should render matrix question 1`] = `
<styled.div>
  <MatrixQuestion
    chartType="foo"
    data={
      Object {
        "barData": Object {},
      }
    }
    isOptionImage={false}
    question={Object {}}
    showLegend={true}
  />
</styled.div>
`;

exports[`Chart component should render open question 1`] = `
<Connect(OpenQuestion)
  data={
    Object {
      "barData": Object {},
    }
  }
  question={Object {}}
/>
`;

exports[`Chart component should render order question 1`] = `
<Connect(DivideQuestion)
  data={
    Object {
      "barData": Object {},
    }
  }
/>
`;
